File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialogs/DialogSetOrderType.java
package com.iamkatrechko.clipboardmanager.view.dialogs;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

import com.iamkatrechko.clipboardmanager.R;
import com.iamkatrechko.clipboardmanager.domain.util.UtilPreferences;

/**
 * Диалог выбора сортировки заметки
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
public class DialogSetOrderType extends DialogFragment {

    /**
     * Возвращает новый экземпляр диалога
     * @return новый экземпляр диалога
     */
    public static DialogSetOrderType newInstance() {
        return new DialogSetOrderType();
    }

    /**
     * Возвращает данные обратно во фрагмент
     * @param orderType тип сортировки
     */
    private void sendResult(String orderType) {
        if (getTargetFragment() == null) {
            return;
        }
        Intent a = new Intent();
        a.putExtra("orderType", orderType);
        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, a);
    }

    @NonNull
    public Dialog onCreateDialog(Bundle bundle) {
        final String[] mItemsNames = getResources().getStringArray(R.array.orders_title);
        final String[] mItemsValues = getResources().getStringArray(R.array.orders_values);
        int currentOrderType = Integer.valueOf(UtilPreferences.getOrderType(getActivity()));

        return new AlertDialog.Builder(getActivity())
                .setTitle("Сортировка:")
                .setSingleChoiceItems(mItemsNames, currentOrderType - 1, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int item) {
                        sendResult(mItemsValues[item]);
                        dismiss();
                    }
                })
                .create();
    }
}

File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialogs/DialogSetOrderType.kt
package com.iamkatrechko.clipboardmanager.view.dialogs

import android.app.Activity
import android.app.AlertDialog
import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.DialogFragment
import com.iamkatrechko.clipboardmanager.domain.param.values.OrderType
import com.iamkatrechko.clipboardmanager.domain.util.UtilPreferences

/**
 * Диалог выбора сортировки заметки
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
class DialogSetOrderType : DialogFragment() {

    /**
     * Возвращает данные обратно во фрагмент
     * @param orderType тип сортировки
     */
    private fun sendResult(orderType: Int) {
        if (targetFragment == null) {
            return
        }
        val intent = Intent()
        intent.putExtra("orderType", orderType)
        targetFragment.onActivityResult(targetRequestCode, Activity.RESULT_OK, intent)
    }

    override fun onCreateDialog(bundle: Bundle?): Dialog {
        val mItemsNames = OrderType.values().mapNotNull { getString(it.nameResId) }.toTypedArray()
        val current = UtilPreferences.getOrderType(activity)

        return AlertDialog.Builder(activity)
                .setTitle("Сортировка:")
                .setSingleChoiceItems(mItemsNames, current.ordinal) { dialog, which ->
                    sendResult(which)
                    dismiss()
                }
                .create()
    }

    companion object {

        /**
         * Возвращает новый экземпляр диалога
         * @return новый экземпляр диалога
         */
        fun newInstance(): DialogSetOrderType {
            return DialogSetOrderType()
        }
    }
}
